### ISSUE TITLE
SoftMax function for output layer

### ISSUE THREAD

**Issue by a1xon** [wagenaartje/neataptic#82]
A SoftMax Function for the output layer would be awesome for any classification NN. I took a look inside your code and I guess it's not that easy to integrate into the activation functions, am I right?
https://en.wikipedia.org/wiki/Softmax_function

**Comment by wagenaartje** [comment]
I have thought about this, your assumption is right. It's quite hard to integrate it as activations are now completely independent of each other. It would also require a more layered structure of networks, which is not the case right now.

**Comment by a1xon** [comment]
So if I choose a linear activation function for the last output nodes and do a softmax on my own this would result in nearly the same NN correct?

**Comment by wagenaartje** [comment]
Yes. That would have the same effect. However, you can't back-propagate this. It would be possible to do evolution with though.
